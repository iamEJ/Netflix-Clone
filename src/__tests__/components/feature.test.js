import React from "react";
import { render } from "@testing-library/react";
import Feature from "../../components/feature";

describe("<Feature /> ", () => {
  it("renders <Feature /> with populated data", () => {
    const { container, getByText } = render(
      <Feature>
        <Feature.Title>Unlimited films, TV shows and more.</Feature.Title>
        <Feature.SubTitle>Watch anywhere. Cancel at any time.</Feature.SubTitle>
      </Feature>
    );
    expect(getByText("Unlimited films, TV shows and more.")).toBeTruthy();
    expect(getByText("Watch anywhere. Cancel at any time.")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders <Feature /> with just a title", () => {
    const { container, getByText, queryByText } = render(
      <Feature>
        <Feature.Title>Unlimited films, TV shows and more.</Feature.Title>
      </Feature>
    );
    expect(getByText("Unlimited films, TV shows and more.")).toBeTruthy();
    expect(queryByText("Watch anywhere. Cancel at any time.")).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
