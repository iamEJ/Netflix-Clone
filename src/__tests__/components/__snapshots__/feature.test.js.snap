// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feature />  renders <Feature /> with just a title 1`] = `
<div
  class="feature__Container-sc-12znx9v-0 kJpSSV"
>
  <h1
    class="feature__Title-sc-12znx9v-1 gXNwBQ"
  >
    Unlimited films, TV shows and more.
  </h1>
</div>
`;

exports[`<Feature />  renders <Feature /> with populated data 1`] = `
<div
  class="feature__Container-sc-12znx9v-0 kJpSSV"
>
  <h1
    class="feature__Title-sc-12znx9v-1 gXNwBQ"
  >
    Unlimited films, TV shows and more.
  </h1>
  <h2
    class="feature__SubTitle-sc-12znx9v-2 htKjJA"
  >
    Watch anywhere. Cancel at any time.
  </h2>
</div>
`;
